src/roiextractors/extractors/memmapextractors/memmapextractors.py

```diff
@@ -64,43 +62,80 @@ def get_num_channels(self):
         return self._num_channels
 
     @staticmethod
-    def write_imaging(imaging_extractor: ImagingExtractor, save_path: PathType = None, verbose: bool = False):
+    def write_imaging(
+        imaging_extractor: ImagingExtractor,
+        save_path: PathType,
+        verbose: bool = False,
+        buffer_gb: int = 0,
+    ):
         """
         Static method to write imaging.
 
         Parameters
         ----------
-        imaging: ImagingExtractor object
-            The EXTRACT segmentation object from which an EXTRACT native format
-            file has to be generated.
+        imaging: An ImagingExtractor object that inherited from MemmapImagingExtractor
         save_path: str
-            path to save the native format.
-        overwrite: bool
-            If True and save_path is existing, it is overwritten
+            path to save the native format to.
+        verbose: bool
+            Displays a progress bar.
+        buffer_gb: int
+            The size of the buffer in Gigabytes. Default of 0 forces chunks that cover only one image.
         """
         imaging = imaging_extractor
-        video_to_save = np.memmap(
+        file_size_in_bytes = Path(imaging.file_path).stat().st_size
+        available_memory_in_bytes = psutil.virtual_memory().available
+        buffer_size_in_bytes = buffer_gb * 10**9
```

Comment:
In line with setting this argument appropriately as a `float`, this should use `1e9` instead of `10**9`

Reply:
Makes sense.

Reply:
Done in the last commit.